chalice/package.py

```diff
@@ -858,17 +857,198 @@ def _generate_managediamrole(self, resource, template):
+        }.get(route_key,
+              'integrations/${aws_apigatewayv2_integration'
+              '.websocket_message_api_integration.id}')
+
+        route_resource_name = {
+            '$connect': 'websocket_connect_route',
+            '$disconnect': 'websocket_disconnect_route',
+            '$default': 'websocket_message_route',
+        }.get(route_key, 'message')
+
+        template['resource'].setdefault(
+            'aws_apigatewayv2_route', {}
+        )[route_resource_name] = {
+            "api_id": websocket_api_id,
+            "route_key": route_key,
+            "target": integration_target
+        }
+        return route_resource_name
+
+    def _add_websocket_domain_name(self, websocket_api_id, resource, template):
+        # type: (str, models.WebsocketAPI, Dict[str, Any]) -> None
+        if resource.domain_name is None:
+            return
+        domain_name = resource.domain_name
+
+        ws_domain_name_definition = {
+            "domain_name": domain_name.domain_name,
+            "domain_name_configuration": {
+                'certificate_arn': domain_name.certificate_arn,
+                'endpoint_type': 'REGIONAL',
+            },
+        }
+
+        if domain_name.tags:
+            ws_domain_name_definition['tags'] = domain_name.tags
+
+        template['resource'].setdefault(
+            'aws_apigatewayv2_domain_name', {}
+        )[domain_name.resource_name] = ws_domain_name_definition
+
+        template['resource'].setdefault(
+            'aws_apigatewayv2_api_mapping', {}
+        )[domain_name.resource_name + '_mapping'] = {
+            "api_id": websocket_api_id,
+            "domain_name": "${aws_apigatewayv2_domain_name.%s.id}" %
+                           domain_name.resource_name,
+            "stage": "${aws_apigatewayv2_stage.websocket_api_stage.id}",
+        }
+
+    def _inject_websocketapi_outputs(self, websocket_api_id, template):
+        # type: (str, Dict[str, Any]) -> None
+        aws_lambda_functions = template['resource']['aws_lambda_function']
+        stage_name = \
+            template['resource']['aws_apigatewayv2_stage'][
+                'websocket_api_stage'][
+                'name']
+        output = template.setdefault('output', {})
+        output['WebsocketAPIId'] = {"value": websocket_api_id}
+
+        if 'websocket_connect' in aws_lambda_functions:
+            output['WebsocketConnectHandlerArn'] = {
+                "value": "${aws_lambda_function.websocket_connect.arn}"}
+            output['WebsocketConnectHandlerName'] = {
+                "value": "${aws_lambda_function.websocket_connect}"}
+        if 'websocket_message' in aws_lambda_functions:
+            output['WebsocketMessageHandlerArn'] = {
+                "value": "${aws_lambda_function.websocket_message.arn}"}
+            output['WebsocketMessageHandlerName'] = {
+                "value": "${aws_lambda_function.websocket_message}"}
```

Comment:
I noticed this prints out all the properties of the lambda function: 

```
WebsocketMessageHandlerName = {
  "architectures" = tolist([
    "x86_64",
  ])
  "arn" = "arn:aws:lambda:us-west-2:12345:function:testws-dev-websocket_message"
  "code_signing_config_arn" = ""
  "dead_letter_config" = tolist([])
  "description" = ""
  "environment" = tolist([])
```

The SAM/cfn version uses `{'Ref': 'WebsocketMessage'}` which just returns the [resource name](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#aws-resource-lambda-function-return-values).

I believe we just need to add a `.function_name` to this output (same to all the other ws function handlers).

Reply:
Nice catch! Thanks for fixing it @jamesls